CMakeLists.txt

```diff
@@ -30,11 +32,14 @@ if(NOT ${PROJECT_NAME}_IS_DEPENDENCY)
 endif()
 # Exclude any embedded CATCH
 list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/CATCH")
-if (QUICKCPPLIB_REQUIRE_CXX20)
+list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/detail/config\\.hpp")
+if (QUICKCPPLIB_USE_SYSTEM_SPAN_LITE OR QUICKCPPLIB_REQUIRE_CXX20)
   list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/span-lite/")
```

Comment:
> purpose of filtering these out, it was because they filled the Visual Studio project file with lots of files which were irrelevant.

Unfortunately that's not the whole story. The `${PROJECT_NAME}_HEADERS` list determines which files are `install(FILES ` ([1]) and these files are also referenced by the generated and installed `quickcpplibExports.cmake` ([2]). So we need to filter these lists for package managers like vcpkg which don't clone and therefore install these submodules. However, always filtering would lead to files missing from the installation.

[1]: https://github.com/ned14/quickcpplib/blob/3a3c21eeeb18b089ebed4fd97795e0afa064f176/cmakelib/QuickCppLibMakeInstall.cmake#L15-L21
[2]: https://github.com/ned14/quickcpplib/blob/3a3c21eeeb18b089ebed4fd97795e0afa064f176/cmakelib/QuickCppLibMakeHeaderOnlyLibrary.cmake#L11-L21